pyquil/paulis.py

```diff
@@ -502,7 +505,7 @@ class PauliSum(object):
     """A sum of one or more PauliTerms.
     """
 
-    def __init__(self, terms):
+    def __init__(self, terms: List[PauliTerm]):
```

Comment:
```suggestion
    def __init__(self, terms: Sequence[PauliTerm]):
```

Reply:
will need to be imported as well -- and this might apply to all the uses of `List` in this file

Reply:
There is already `from collections.abc import Sequence`, so I will do instead `from typing import Sequence as SequenceT`.